In [6]:
import requests
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import yfinance as yf

In [7]:
from datetime import datetime, timedelta
from concurrent.futures import ThreadPoolExecutor, as_completed
from tqdm import tqdm
from io import StringIO
from statsmodels.tsa.stattools import adfuller

Проверить на стационарность ряд цены на нефть (https://finance.yahoo.com/quote/BZ%3DF/history?
period1=1185753600&period2=1666310400&interval=1d&filter=history
&frequency=1d&includeAdjustedClose=true).

In [8]:
ticker = "BZ=F"
start_date = "2022-12-07"
end_date = "2023-12-07"
brent_df = yf.download(ticker, start=start_date, end=end_date)

[*********************100%%**********************]  1 of 1 completed


Проводим расширенный тест Дики-Фуллера. Сформируем гипотезы:
Нулевая гипотеза (H0) предполагает наличие единичного корня - временной ряд нестационарен.
Альтернативная гипотеза (H1) предполагает отсутствие единичного корня - временной ряд стационарен.
Статистика ADF рассчитывается на основе авторегрессионного уравнения. Если статистика меньше критического значения, нулевая гипотеза отвергается.
В тесте будет рассчитано p-value. Установим уровень значимости 0.05 и будем отвергать нуль гипотезу при превышении p-value этого уровня.

In [9]:
adf_test = adfuller(brent_df ['Close'])
adf_test

(-2.305585912161266,
 0.17017273237575237,
 0,
 251,
 {'1%': -3.4566744514553016,
  '5%': -2.8731248767783426,
  '10%': -2.5729436702592023},
 901.9474330849062)


Мы получили статистику -2.3, что меньше критического значения для 5% уровня значимости. p-value 0.17 > 0.05
Таким образом, не получилось отвергнуть нуль гипотезу, временной ряд нестационарен